### PR TITLE
Block streaming timeouts

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Improve error messages for block streaming in the transport layer. Change the streaming timeout from fixed 10 minutes to a dynamic value based on the block size.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3017

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: comm/src/main/scala/coop/rchain/comm/UPnP.scala [hunk 1]
```diff
@@ -133,7 +133,7 @@ object UPnP {
     for {
       _       <- Log[F].info("trying to open ports using UPnP....")
       devices <- Sync[F].delay(discover)
-      res <- if (devices.gateways.isEmpty) logGatewayEmpty(devices) *> None.pure[F]
+      res <- if (devices.gateways.isEmpty) logGatewayEmpty(devices) >> None.pure[F]
```
Comment (tymm) [inline comm/src/main/scala/coop/rchain/comm/UPnP.scala]: What's the reason for the change to `>>`?

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/UPnP.scala]: @tymm Because Applicative `productR` a.k.a `*>` is considered dangerous. It can lead to memory leaks and/or race conditions. However, this change is not related to the PR ticket. I used the opportunity to clean up and replace `productR` with `flatMap` in the `comm` layer.

(see: https://twitter.com/impurepics/status/1094339310571175937)
